### PR TITLE
Specify clearOn options for platform services

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Uses `UserKeyDefinitions` where possible within platform code to utilize `clearOn` events and in preparation to remove interop between `KeyDefinitions` and `UserKeyDefinitions`

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: libs/common/src/platform/services/config/default-config.service.ts [hunk 1]
```diff
@@ -32,7 +32,6 @@ export const USER_SERVER_CONFIG = new UserKeyDefinition<ServerConfig>(CONFIG_DIS
   clearOn: ["logout"],
 });
 
-// TODO MDG: When to clean these up?
```
Comment (MGibson1) [inline libs/common/src/platform/services/config/default-config.service.ts:35]: We are not clearing global-level configs

File: libs/common/src/platform/services/key-state/org-keys.state.ts [hunk 2]
```diff
@@ -4,13 +4,14 @@ import { OrganizationId } from "../../../types/guid";
...
-import { KeyDefinition, CRYPTO_DISK, DeriveDefinition } from "../../state";
+import { CRYPTO_DISK, DeriveDefinition, UserKeyDefinition } from "../../state";
 
-export const USER_ENCRYPTED_ORGANIZATION_KEYS = KeyDefinition.record<
+export const USER_ENCRYPTED_ORGANIZATION_KEYS = UserKeyDefinition.record<
   EncryptedOrganizationKeyData,
   OrganizationId
 >(CRYPTO_DISK, "organizationKeys", {
   deserializer: (obj) => obj,
+  clearOn: ["logout"],
```
Comment (justindbaur) [inline libs/common/src/platform/services/key-state/org-keys.state.ts:14]: Was there other code that was deleting this on logout? Should we delete that code?

File: libs/common/src/platform/services/key-state/provider-keys.state.ts [hunk 3]
```diff
@@ -3,14 +3,15 @@ import { ProviderKey } from "../../../types/key";
...
+import { CRYPTO_DISK, DeriveDefinition, UserKeyDefinition } from "../../state";
 import { CryptoService } from "../crypto.service";
 
-export const USER_ENCRYPTED_PROVIDER_KEYS = KeyDefinition.record<EncryptedString, ProviderId>(
+export const USER_ENCRYPTED_PROVIDER_KEYS = UserKeyDefinition.record<EncryptedString, ProviderId>(
   CRYPTO_DISK,
   "providerKeys",
   {
     deserializer: (obj) => obj,
+    clearOn: ["logout"],
```
Comment (justindbaur) [inline libs/common/src/platform/services/key-state/provider-keys.state.ts:14]: Same, can the other code being used to clear this be removed?

File: libs/common/src/platform/services/key-state/user-key.state.ts [hunk 4]
```diff
@@ -3,18 +3,25 @@ import { CryptoFunctionService } from "../../abstractions/crypto-function.servic
...
   deserializer: (obj) => obj,
 });
 
-export const USER_ENCRYPTED_PRIVATE_KEY = new KeyDefinition<EncryptedString>(
+export const USER_ENCRYPTED_PRIVATE_KEY = new UserKeyDefinition<EncryptedString>(
   CRYPTO_DISK,
   "privateKey",
   {
     deserializer: (obj) => obj,
+    clearOn: ["logout"],
```
Comment (justindbaur) [inline libs/common/src/platform/services/key-state/user-key.state.ts:24]: Same question.

File: libs/common/src/platform/services/key-state/user-key.state.ts [hunk 5]
```diff
@@ -58,6 +65,7 @@ export const USER_PUBLIC_KEY = DeriveDefinition.from<
     return (await cryptoFunctionService.rsaExtractPublicKey(privateKey)) as UserPublicKey;
   },
 });
-export const USER_KEY = new KeyDefinition<UserKey>(CRYPTO_MEMORY, "userKey", {
+export const USER_KEY = new UserKeyDefinition<UserKey>(CRYPTO_MEMORY, "userKey", {
   deserializer: (obj) => SymmetricCryptoKey.fromJSON(obj) as UserKey,
+  clearOn: ["logout", "lock"],
```
Comment (justindbaur) [inline libs/common/src/platform/services/key-state/user-key.state.ts:70]: Here too

File: libs/common/src/platform/services/key-state/org-keys.state.ts (hunk 2 above)
Comment (MGibson1) [reply libs/common/src/platform/services/key-state/org-keys.state.ts:14]: yup, and we sure can, now!

File: libs/common/src/platform/biometrics/biometric.state.ts
Comment (Hinton) [inline libs/common/src/platform/biometrics/biometric.state.ts:1]: Shouldn't this clear on logout?

Comment (MGibson1) [reply libs/common/src/platform/biometrics/biometric.state.ts:1]: Not for most of them. User settings are generally persisted through logout. [Here is state service's implementation that impacts this at a global level](https://github.com/bitwarden/clients/blob/22f04195778d7d034c1cb8aca6c215b6520bff8e/libs/common/src/platform/services/state.service.ts#L1516-L1525)

I'm fine with changing this behavior, but I don't think that should be done as part of this work, which is just trying to enable removing method overrides we don't want to support. That shouldn't be bogged down by these product questions.

**Review by justindbaur — APPROVED** [review]

**Review by Hinton — APPROVED** [review]
LGTM, I do think the biometric unlock enabled setting should be revisited though. Not sure what side effects having that setting without a key has.

**Pull request merged**